src/scripts/client/modules/gradebook.js

```diff
@@ -373,11 +373,14 @@
                       }
                   }
                 } else {
-                  console.log("Submission from other path", submission, students[ studentId ])
+                  console.log('Submission from other path', submission, students[ studentId ]);
                 }
             } );
 
         }
+        if (sessionData.students.length === 0) {
+          throw new Error('no students');
```

Comment:
Since our `scrape` function only has a success callback and it is already wrapped in a try catch , this throw allows us to bubble up to our failure handler in that catch block. While the exact message is not caught, all errors are swallowed if something in scrape fails.   If we just `return;` we will need to add additional structure to handle the void return to update the `$ui` versus leveraging our existing failure handler.

Catch is here. https://github.com/TIYDC/tiyo-assistant/pull/54/files/a1eed9a2e977c6cb8379a01e733368152fb4bc5b#diff-524605ea9ca4e46db2124c9210cbe7b7L138